### ISSUE TITLE
live reload is removed from VS marketplace.

### ISSUE THREAD

**Issue by anirudhagupta** [xamarin/Xamarin.Forms#4155]
Today I search the VS marketplace, can't find the extension. this has reported earlier on https://github.com/MicrosoftDocs/xamarin-docs/issues/927

Is there any workaround for this, I was looking for using it for UWP, Any workaround.

**Comment by ysmoradi** [comment]
UWP's XAML designer is great. You can use Expression blend to modify that too. You can also create amazing animations there! It's really fast and mature. **But they do provide xaml edit & continue at runtime too**. Because **they know there is a difference between designer and runtime edit & continue**. You've gotta kidding me! Please do not remove the android apk publishing tool because you want to provide a xaml designer tool! We really need those stuff!

**Comment by FetFrumos** [comment]
Why? This is a very strange decision. This is the only free tool that worked.
This is a real need, this is what you need out of the box. Why aren't you interested in community opinion? I spend the days of my life waiting for recompilation :(.

**Comment by cesarcamps** [comment]
Very strange, indeed. Has a reminiscence of the old microsoft.

**Comment by tlhintoq** [comment]
What an awesome tool... The ability to just hit 'save' and watch 3 different platform devices update at the same time!  

Yet another XAML previewing tool started and discarded by the MS-Xamarin department.  All the man-hours wasted that could have gone towards making any one of them work.  Or fixing any of the other hundred logged issues.

**Comment by dansiegel** [comment]
@pauldipietro I cannot emphasize enough that not moving forward with Live Reload is a **MAJOR** mistake! The Previewer has certainly gotten better, but it's extremely slow, and will always be plagued by a variety of problems that Live Reload never has to worry about since you're working with your actual app's code. 

- As an example the Previewer requires you to either have all of your fields initialized to properly display or bloat your code with checks to see if you're in design mode. But this is and always will be a static view.
- Live Reload on the other hand simply let's you run your app with an edit & continue mentality that is far more useful. This means that as the developer is updating my XAML, they can see Animations, add Behaviors, Gestures and Effects, and then validate that these elements are working. Not to mention that this then allows the developer to make changes in the UI that update State, etc. 

Support for Live Reload needs to continue, and frankly it needs to be baked in as part of the standard Xamarin tooling in VS2019/VSMac8

cc: @davidortinau @migueldeicaza @pierceboggan

**Comment by cesarcamps** [comment]
I can not believe this! Why!!!?????

**Comment by cesarcamps** [comment]
And in the gitter, nobody is answering
https://gitter.im/xamarin/live-reload

**Comment by FetFrumos** [comment]
We are developers  - this is our time, respectively - our money. Xamarin Team - please explain the reasons for this decision. Transfer the LiveReload to the open source? if you do not have the resources for further development. The community needs this free tool.
Silence is a bad strategy.

**Comment by tlhintoq** [comment]
> 
> 
> And in the gitter, nobody is answering
> https://gitter.im/xamarin/live-reload

What is 'gitter'?  Please don't tell me that Xamarin has obfuscated the source of where they want developer feedback again... yet again... another time and another social media platform....

**Comment by cesarcamps** [comment]
Gitter is the source, where live-reload developers interact with developers. Or at last until last week.

**Comment by cesarcamps** [comment]
Excuse my language, english is not my mother lang.

**Comment by cesarcamps** [comment]
I began to lost faith in xamarin. I used many years now, from where was Mono.Android/IOS, through Xamarin name changed, to Microsoft's bought. I poured many, many, many, days and nights in it. And now see, my choices was erronous :(. It is a very very sad days.
But the signs are all here around us. I choose to not see it.

**Comment by tlhintoq** [comment]
> 
> 
> Hummm, sorry about that :(
> In the comment sections, or something, Daniel Cazzulino @kzu, say that through this channel, the communication will be more streamlined.
> And if you enter, you see messages back and forth between live-reload developers (majorly kzu ), with others.

I don't doubt there was talk there.  My point is that either Xamarin or Microsoft does this every so often.  Every team has a different feedback channel - and they like to change them every time the list of problems gets so long that someone's bonus check is in jeopardy.

If you don't know the secret handshake to get an invitation to the hidden chat room then you're just out in the cold.  Which is about 99% of us.  Clearly few people knew about the Gitter room or they wouldn't all be asking _"So does anyone know?"_ on the Xamarin Forum site.

**Comment by cesarcamps** [comment]
Indeed!! I am very disappointed with Microsoft in this matter.
Very disappointed!!
And my apps are in this technologies (Xamarin.Android, Xamarin.IOS) and my company are in the migration process to Xamarin.Forms. And when i see this, i am like W.T.F.??????? Sorry by this language i am very piss of.

**Comment by tlhintoq** [comment]
> 
> 
> Indeed!! I am very disappointed with Microsoft in this matter.
> Very disappointed!!
> And my apps are in this technologies (Xamarin.Android, Xamarin.IOS) and my company are in the migration process to Xamarin.Forms. And when i see this, i am like W.T.F.??????? Sorry by this language i am very piss of.

I agree 100% - No reason to apologize.   "WTF?!" is the right answer.  So long as you don't use that abbreviation on the Xamarin Forum.  I was scolded by management for it.  Really.

Same boat as you.  I pushed my company to migrate to Xamarin.  Now I feel tricked.  Its not about the one technology of LiveReload.  I get that any one thing can be a failed path.  Its the clear and obvious pattern of this type of activity that has gotten to me.  One has to be involved long enough to see the pattern emerge.  Then once you're so invested that you can see the pattern, its too late because you are too deeply invested to get out.  Like marriage.

**Comment by cesarcamps** [comment]
Breaking news!!! 

Daniel Cazzulino @kzu 15:49
I don't know what's going on, I dont
manage that marketplace account. I haven't been notified of anything so far regarding this. So I can't provide any further info, sorry 

https://gitter.im/xamarin/live-reload

**Comment by cesarcamps** [comment]
https://gitter.im/xamarin/Xamarin.Forms?at=5bcdfb2938449236617e65e8
Read this, Icaza's words

**Comment by cesarcamps** [comment]
In other words, we are screwed. Very hard!!

**Comment by cesarcamps** [comment]
If i have to pay for a similar tool, ex LIveXAML, i rather try  to recode all my apps on flutter. Mark my words!!

**Comment by pictos** [comment]
@tlhintoq can you share the installer?

**Comment by tlhintoq** [comment]
@pictos 
That's a good question.  I'll have to find it first.  Major project with that damned Windows Marketplace is that everything is hidden and behind the scenes.  You don't get a .msi file these days that you can run.  I'll dig and see what I can find.

**Comment by dansiegel** [comment]
@pauldipietro I should bring #3921 to your attention as well, as it really is a point in case for why reliance on the Previewer is foolish at best. There are simply too many scenarios in real application development that may break the previewer. Not to mention as I discussed with @pierceboggan, the fact that default styles are radically different between platforms (i.e. Buttons look more like a link on iOS but actually like a button on Android), not to mention that we may be implementing custom Idiom/Platform styling where really the Previewer is being asked to try to figure out too much that the best solution becomes run your real app with edit/continue.

**Comment by mackayn** [comment]
This is a dreadful decision, two fingers up to the Forms Dev community, we've been battling with sub standard tooling in VS2017 (especially the woeful 15.8), decisions like this and keeping the profiler as Enterprise not based on common sense.

**Comment by pictos** [comment]
@pauldipietro i don't know about the future plans to Xamarin and i great fan about LiveReload (LR), the team made a great job! My productivity "went to the heights".  It's certain the LiveReload will be discontinued, to give way to Previewer. But until this date, would not it be possible to keep LR available? 

And being a bit more hopeful, would it be possible to make LR open source?

**Comment by tlhintoq** [comment]
> If previewer is intended to absorb code from LiveReload and become the best of both - then great. **WHEN IT IS DONE AND WORKING** as a replacement then do so. But to kill off LiveReload when you have nothing in place to fill the void is just... irresponsible to those of us actually trying to earn a living with these tools.

**Comment by mackayn** [comment]
> there is this open source livereload project but i dont know how good it is. when i see the source code looked to me already complicated :). i would love to contribute on it if it is promising https://github.com/ylatuya/XAMLator
> 
> I am guessing that there is some legal issue between livereload and livexaml. thats why Microsoft discontinued it. if it is not legal issue, why would they do it for no reason, even they plan to build better previewer, it wont bite anybodies *ss if it remains as it can be even open source.

Can't imagine the legal issue, MS have the IP for Forms & XAML. It's hardly a patent as MS have been doign it since 2005/8.

Previewer was demo'd at Evolve 2016, I was there, It never worked since and only 'just' works in 2018 for basic projects, hot loading...nope etc etc. So we should base our optimism on?

**Comment by brianlagunas** [comment]
@pauldipietro That probably should have been a new comment instead of an update to an existing comment that had so many down votes. Plus, no one got notification of the update.

**Comment by jamesmontemagno** [comment]
I want to thank everyone for your feedback and comments. Xamarin Live Reload was an experimental technology that gave us great insight into developer productivity into different scenarios for UI design. At this time we are taking a step back from any of our experimental technologies and are refocusing our efforts to stabilize and improve great productivity experiences already built into Visual Studio with our XAML Preview and in IDE tooling such as XAML IntelliSense, faster Android build times, and more. Please read through our roadmap features for Visual Studio 2019 to learn more about new productivity features that we are working on for Xamarin.Forms. The extension has been relisted and will remain available for Visual Studio 2017, but will not be receiving future updates.

**Comment by tlhintoq** [comment]
CC: @jamesmontemagno @brianlagunas 
Well... A step in the right direction.  At least the best of the two technologies (LiveReload versus Previewer) is once again available to people _while the new technologies are in development_.  That makes so much more sense than pulling them with no replacement yet available.

**I'll further applaud** re-focusing on stability and fixing the current state of... everything... to get to a point where everything that currently exists can be trusted and used with confidence.  I long for a day when a simple `label` works as expected or where the state of the UI doesn't revert after a navigation change and return.

I think its a shame that a company the size of Microsoft is basically saying:
>We _won't_ allocate enough human resources to both fix existing things **and** improve the developer tooling.
But I get that all companies have to draw a line someplace.  Everything comes down to a line in the sand as to where they will spend money.  I don't have to like it.  I don't see that MS can't afford to do both.  But I can see that the answer isn't always in piling on more people.  Things do come a point where you can't have 47 people all working on the same files at the same time without stomping on each other's work.

I'm not sure how I feel about the phrasing "coming in Visual Studio 2019".  Many of us in enterprise situations have only recently gotten to move to VS2017-Which is still buggy as all get out.  Unlike individual freelancers we can't jump onto the `v.today` of most tools.  I'm trying to **not** read into this 2019 comment as 
>"Don't expect improvements in VS2017 which most of you are reporting countless problems with.
But also not seeing another way to take that.

I would like to ask: Is there Xamarin Roadmap someplace?  Because if all the developer support is being pulled off the tooling like LiveReload and being re-allocated to bug fixing Xamarin then we (the developer community) should therefore be expecting big things and faster version updates of the Xamarin eco-system - right?

**Comment by FetFrumos** [comment]
I have user case when LiveReload worked but LiveXaml not worked. This is a very useful tool. I propose to develop it in the community. Need open source. @jamesmontemagno @brianlagunas - this is possible?

**Comment by pellet** [comment]
Funny this isn't a priority, there is a clear distinction between flutter/react native's live-reload capabilities and xamarin's lack thereof.  This preview version actually worked quite well, I don't understand why it was dropped.

**Comment by FunkyLambda** [comment]
Hey @jamesmontemagno, does the Previewer currently offer all the features that were already on offer with LiveReload? If not, it’s an inferior product and should be ditched for LiveReload. It should then be LiveReload that comes standard as part of VS.
It’s either a case of:
1) You don’t have enough pull within MS to make it happen;
2) You are not aware of what devs actually want, in which case, see above;
3) You actually belive there’s hope for the Previewer.

**Comment by EmilAlipiev** [comment]
@FunkyLambda no, it doesnt offer anything yet. it cant even display a proper scrolview. Actually idea of previewer is good as i can understand. if it works very well, you can simply test your ui on different phone sizes like you could do with react native on the browser. But I doubt that it is easily doable in a short term. we have to also create testable static lists (see James' presentation on VS launch event). I would love to have previewer as a 2nd tool, maybe to test some UI unit tests or some static pages. But we surely need a hot reload tool like live reload. 
There are bunch of open source applications but all of them has something missing or some serious problems. Livereload was the best one working. At least if xamarin doesnt want to continue with it, they can open source it. Thus we could all contribute to develop it further.

**Comment by AndreiMisiukevich** [comment]
For now everybody may use free opensource HotReload https://github.com/AndreiMisiukevich/HotReload

**Comment by slang25** [comment]
^ I've tried a few solutions, paid for and open source. This is my favourite tool today by far, plugins for VS, VS for Mac, Rider and works from the command line.

**Comment by john-larson** [comment]
I have been informed that the proper place for requesting features and voting for them is on https://developercommunity.visualstudio.com. (I guess this is where the Xamarin team looks to decide what the community wants.) So I went ahead and created a feature request for Live Reload: https://developercommunity.visualstudio.com/idea/618119/xamarinforms-live-hot-reload.html. Please feel free to vote for the feature if you request it too.
